### PR TITLE
FRONT-2476: Form input group created.

### PR DESCRIPTION
**Pull request by Maxfire**
Jira issue(s):
- https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-413

### CODE REVIEW

File: templates/patterns/form_input/form_input.ui_patterns.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+form_input:
+  label: "Form input"
+  description: "Form input. https://getbootstrap.com/docs/5.0/forms/input-group/"
+  variants:
+    default:
+      label: Text
+      description: Default input text.
+    checkbox:
+      input_type: checkbox
```
Comment (escuriola) [inline templates/patterns/form_input/form_input.ui_patterns.yml]: Not needed

File: templates/patterns/form_input/form_input.ui_patterns.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      label: Checkbox
+      description: Default checkbox input.
+    file:
+      input_type: file
```
Comment (escuriola) [inline templates/patterns/form_input/form_input.ui_patterns.yml]: Not needed

File: templates/overrides/form/input--textfield.html.twig [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#}
+
+  {%
+    set classes = [
+      'form-control',
+       attributes.hasClass('error') ? 'is-invalid',
+    ]
+  %}
+{% if attributes.placeholder is empty %}
+  {% set attributes = attributes.setAttribute('placeholder', 'text here')  %}
```
Comment (escuriola) [inline templates/overrides/form/input--textfield.html.twig]: This default text should be translatable

**Review by escuriola — COMMENTED** [review]
Some styles missing, for instance form-required has not any special style to mark a field as required, and checkbox left-margin doesn't fit well sometimes.
I'm not sure this must be reviewed now or later in a general form/page theming.

**Review by escuriola — DISMISSED** [review]

File: templates/overrides/form/form-element--checkbox.html.twig [hunk 4]
```diff
@@ -16,7 +16,7 @@
     title_display not in ['after', 'before'] ? 'form-no-label',
     disabled == 'disabled' ? 'form-disabled',
     errors ? 'form-item--error',
-    'form-check',
+    switch ? 'form-check form-switch': 'form-check',
```
Comment (drishu) [inline templates/overrides/form/form-element--checkbox.html.twig]: so we do this in checkbox, if needed

File: templates/overrides/form/input--textfield.html.twig [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ * Available variables:
+ * - attributes: A list of HTML attributes for the input element.
+ * - children: Optional additional rendered elements.
+ *
+ * @see template_preprocess_input()
+ */
+#}
+
+  {%
```
Comment (drishu) [inline templates/overrides/form/input--textfield.html.twig]: no intend needed

File: templates/overrides/form/input--textfield.html.twig [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    set classes = [
+      'form-control',
+       attributes.hasClass('error') ? 'is-invalid',
+    ]
+  %}
+{% if attributes.placeholder is empty %}
+  {% set attributes = attributes.setAttribute('placeholder', 'text here'|t)  %}
```
Comment (drishu) [inline templates/overrides/form/input--textfield.html.twig]: remove this

**Review by drishu — CHANGES_REQUESTED** [review]
- the helper text has the correct class, and even though not a div, visually its ok
- valid-feedback and invalid-feedback are handled differently in drupal so we won't touch those
- if validation error is present the message is presented on top in the messages area and the border is collored red as it should be (in netlify its not and that is a mistake I feel)

**Review by drishu — DISMISSED** [review]

File: templates/overrides/form/form-element-label.html.twig [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+ * @see template_preprocess_form_element_label()
+ */
+#}
+{%
+  set classes = [
+  title_display == 'after' ? 'option',
+  title_display == 'invisible' ? 'visually-hidden',
+  required ? 'js-form-required',
+  required ? 'form-required',
+  'form-label',
```
Comment (brummbar) [inline templates/overrides/form/form-element-label.html.twig]: Nit: these are to be indented.

File: templates/overrides/form/input--textfield.html.twig [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+ * @see template_preprocess_input()
+ */
+#}
+{%
+  set classes = [
+    'form-control',
+     attributes.hasClass('error') ? 'is-invalid',
+  ]
+%}
+
```
Comment (brummbar) [inline templates/overrides/form/input--textfield.html.twig]: Nit: Drop this empty line, it generates un unneeded empty line.

File: templates/overrides/form/input--textfield.html.twig [hunk 9]
```diff
@@ -0,0 +1,20 @@
+{#
+/**
+ * @file
+ * Theme override for an 'input' #type form element.
```
Comment (brummbar) [inline templates/overrides/form/input--textfield.html.twig]: This is not correct, this is an override for input of type textfield.

**Review by brummbar — CHANGES_REQUESTED** [review]

**Review by brummbar — APPROVED** [review]

**Review by drishu — APPROVED** [review]

**Pull request merged**